### ISSUE TITLE
Does course-dl resume downloads?

### ISSUE THREAD

**Issue by madmax2** [coursera-dl/coursera#360]
Does course-dl resume downloads from where you left off?

e.g. If I download upto section 3, video 1 and it froze at 91%

If I run the script again, 
Will it start the download at section3, video 1 at 0% or at 91%?

Or will it redownload section 1 from beginning again?

**Comment by shivams** [comment]
The short answer is **NO**. 

When you have downloaded some content previously, `coursera-dl` will automatically detect it and skip it. However, there is a caveat. 

When the download of a video is going on and you interrupt the download using <kbd>Ctrl</kbd>+<kbd>C</kbd>, then `coursera-dl` will delete the partially downloaded file. However, if your download process is killed due to some external effects, say your PC crashed, then the partially downloaded file is not deleted. And when you run `coursera-dl` the next time, it will think think that the partially downloaded file is a completely downloaded file. 

In short, `coursera-dl` has no way to identify a partially downloaded file, AFAIK.

**Comment by rbrito** [comment]
Just for the record, @mxamin has been working on this exact issue via his pull request #378 and it is going to be merged very soon.

**Comment by rbrito** [comment]
Hi.

We now have this feature on our current pre-release. To make use of it, please install with
```
pip install --pre coursera
```

If you omit the `--pre` there, it will not get the pre-release version that has this feature. Feedback (works? doesn't work?) is highly welcome.
